ext/ldap/ldap.c

```diff
@@ -2291,20 +2291,20 @@ static void php_ldap_do_modify(INTERNAL_FUNCTION_PARAMETERS, int oper, int ext)
 			num_berval[i] = num_values;
 			ldap_mods[i]->mod_bvalues = safe_emalloc((num_values + 1), sizeof(struct berval *), 0);
 
-			for (j = 0; j < num_values; j++) {
-				zval *ivalue = zend_hash_index_find(Z_ARRVAL_P(value), j);
-				ZEND_ASSERT(ivalue != NULL);
-				convert_to_string(ivalue);
+			zend_ulong attribute_value_index = 0;
+			zval *attribute_value = NULL;
+			ZEND_HASH_FOREACH_NUM_KEY_VAL(Z_ARRVAL_P(value), attribute_value_index, attribute_value) {
+				convert_to_string(attribute_value);
 				if (EG(exception)) {
-					num_berval[i] = j;
+					num_berval[i] = (int)attribute_value_index;
```

Comment:
int may cause problems, uint32_t certainly won't.

Reply:
Fixed in a follow-up commit.